src/main/java/gregtech/common/pipelike/fluidpipe/tile/TileEntityFluidPipeTickable.java

```diff
@@ -122,7 +124,16 @@ private void distributeFluid(int channel, FluidTank tank, FluidStack fluid) {
             IFluidHandler pipeTank = tank;
             CoverBehavior cover = getCoverableImplementation().getCoverAtSide(facing);
             if (cover != null) {
-                pipeTank = cover.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, pipeTank);
+                IFluidHandler capability = cover.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, pipeTank);
+                // Shutter covers return null capability when active, special check to prevent NPE
+                if (cover instanceof CoverShutter && capability == null) {
```

Comment:
This adds hard coupling and that is not something we want as we would need to be considered about it in future.